chain/network/src/peer_manager/tests.rs

```diff
@@ -586,3 +586,25 @@ async fn connection_spam_security_test() {
         c.handshake(&clock.clock()).await;
     }
 }
+
+#[tokio::test]
+async fn outbound_loop_connection() {
+    init_test_logger();
+    let mut rng = make_rng(921853233);
+    let rng = &mut rng;
+    let mut clock = time::FakeClock::default();
+    let chain = Arc::new(data::Chain::make(&mut clock, rng, 10));
+
+    let pm = peer_manager::testonly::start(
+        clock.clock(),
+        near_store::db::TestDB::new(),
+        chain.make_config(rng),
+        chain.clone(),
+    )
+    .await;
+    let mut cfg = chain.make_config(rng);
+    cfg.node_key = pm.cfg.node_key.clone();
+
+    // Starting an outbound loop connection should be stopped without sending the handshake.
```

Comment:
Do we want to test a case of inbound loop connection? If someone steals our private key for signing that's bad, but thats phisically possible and might make sense to test for.

Reply:
done